agent/structs/config_entry_discoverychain_test.go

```diff
@@ -1099,6 +1099,28 @@ func TestServiceRouterConfigEntry(t *testing.T) {
 			}),
 			validateErr: "cannot make use of PrefixRewrite without configuring either PathExact or PathPrefix",
 		},
+		////////////////
+		{
+			name: "route with method matches",
+			entry: makerouter(routeMatch(httpMatch(&ServiceRouteHTTPMatch{
+				Methods: []string{
+					"get", "POST", "dElEtE",
+				},
+			}))),
+			check: func(t *testing.T, entry *ServiceRouterConfigEntry) {
+				m := entry.Routes[0].Match.HTTP.Methods
+				require.Equal(t, []string{"GET", "POST", "DELETE"}, m)
+			},
+		},
+		{
+			name: "route with method matches repeated",
+			entry: makerouter(routeMatch(httpMatch(&ServiceRouteHTTPMatch{
+				Methods: []string{
+					"GET", "DELETE", "get",
+				},
+			}))),
+			validateErr: "Methods contains \"GET\" more than once",
```

Comment:
I'm on the fence about whether we should be lenient here. It doesn't feel like there is much ambiguity about intent or room for footguns if we just allow this. But I'm happy to leave it strict for now as it's easier to start strict and relax if people have legitimate pain from it (e.g. because they concat config from a few sources and it's painful to normalise case and de-dupe for some reason). Seems solvable externally anyway.